### PR TITLE
[ENG-27003] fix: Added pagination and especific args

### PR DESCRIPTION
**Pull request by gomes-fdr**
# [ENG-27003](https://aziontech.atlassian.net/browse/ENG-27003)

## fix: Added pagination and especific args

[ENG-27003]: https://aziontech.atlassian.net/browse/ENG-27003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by azionhsantos — DISMISSED** [review]

File: edgefirewall.yaml [hunk 1]
```diff
@@ -518,12 +549,66 @@ components:
...
+      type: object
       properties:
         name:
           type: string
           enum: [deny, drop, set_rate_limit, set_waf_ruleset, run_function, tag_event]
-        target:
+        argument:
+          oneOf:
+            - $ref: '#/components/schemas/SetRateLimitDetails'
+            - $ref: '#/components/schemas/SetWAFRuleSetDetails'
```
Comment (trgalho) [inline edgefirewall.yaml:560]: **suggestion:** we may need to add a generic Behavior as we not defined the specific schema to all other possible behaviors

```suggestion
            - $ref: '#/components/schemas/SetWAFRuleSetDetails'
            - $ref: '#/components/schemas/GenericBehaviorDetails'  # TBD
```

Comment (gomes-fdr) [reply edgefirewall.yaml:560]: We define both complex details(rate_limit and waf).
Why do we need to provide the simple one?
It just a behavior with the name property.

Comment (gomes-fdr) [reply edgefirewall.yaml:560]: We didn't block any behavior name, when we use set_rate_limit or set_waf_ruleset we have a specific detail, that's all.

Comment (gomes-fdr) [reply edgefirewall.yaml:560]: I think a got your point... the key word is "oneOf" - you are right!

Comment (trgalho) [reply edgefirewall.yaml:561]: Sorry, I arrived here too late. But you got the point.

File: edgefirewall.yaml [hunk 2]
```diff
@@ -518,12 +549,73 @@ components:
...
+          name: set_waf_ruleset
+          argument:
+            waf_id: 1
+            mode: Learning
+            
+    GenericBehaviorDetails:
+      type: object
+      properties:
+          name:
+            type: string
```
Comment (trgalho) [inline edgefirewall.yaml:583]: **todo:** missing "argument" key

The argument key should be defined =/

**Review by azionhsantos — APPROVED** [review]

**Review by trgalho — APPROVED** [review]

**Pull request merged**